api/src/main/java/org/apache/flink/agents/api/Event.java

```diff
@@ -48,4 +48,20 @@ public Object getAttr(String name) {
     public void setAttr(String name, Object value) {
         attributes.put(name, value);
     }
+
+    /**
+     * Get the event type name. The event type name should be the fully qualified class name of the
+     * event, whether it is a Java or Python event.
+     */
+    public String getEventType() {
+        return this.getClass().getCanonicalName();
+    }
+
+    public boolean isInputEvent() {
+        return false;
+    }
+
+    public boolean isOutputEvent() {
+        return false;
+    }
```

Comment:
Doesn't make sense to have these methods here. It introduce the chance for subclasses to incorrectly override these methods. I'd suggest to make the type of input/output events static constants, and compare in `EventMessage`.